modules/client/test/integration/client/encoding.test.ts

```diff
@@ -87,21 +85,19 @@ describe("Client", () => {
           permission: Permissions.UPGRADE_PERMISSION,
         },
       ];
-      let actions: DaoAction[] = [];
+      let actions: IEncodingResult[] = [];
```

Comment:
Why demoving DAOAction? DAO actions are what developers submit on a proposal, encoded data is a different idea/concept.

Reply:
that is not correct. A plugin developer can choose what data it should need in a proposal.
The correct way is to return an "UnsignedTransaction" which could be either executed by the user in the code, by a gas station, can be given to a plugin to execute or used by the `execute()` function in the DAO.

Reply:
Right, but UnsignedTransactions have many more parameters than the ones we need, isn't it? gasPrice, gasLimit, nonce, chainId, .... We only care about the `{to, value, data}`, and this is what the smart contracts receive and send to the `execute` function. 

The `DaoAction` feels like a global type that everybody will use, whereas `IEncodingResult` is a generic concept that could apply to anything, even a base64 representation of a protobuf payload